### PR TITLE
Updated Thymeleaf processors

### PR DESCRIPTION
**Pull request by MarekB01**
https://github.com/BroadleafCommerce/QA/issues/4962

updated thymeleaf processors which were extended from deprecated processor

### CODE REVIEW

**Review by troian88 — COMMENTED** [review]
Can you attach an example of usage and how UI looks with that processors?

File: src/main/java/org/broadleafcommerce/menu/processor/AbstractMenuProcessorExtensionHandler.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public abstract class AbstractMenuProcessorExtensionHandler extends AbstractExte
         implements MenuProcessorExtensionHandler {
 
     @Override
-    public ExtensionResultStatusType addAdditionalFieldsToModel(String tagName, Map<String, String> tagAttributes, Map<String, Object> newModelVars, BroadleafTemplateContext context) {
+    public ExtensionResultStatusType addAdditionalFieldsToModel(Map<String, Object> newModelVars, String menuId, String menuName) {
```
Comment (oleksiimiroshnyk) [inline src/main/java/org/broadleafcommerce/menu/processor/AbstractMenuProcessorExtensionHandler.java]: this is just a stub, there is a real implementation in Theme module I don't see PR for the theme module.
I don't see why would it need menu id as current impl is interested in menuName only.

File: src/main/java/org/broadleafcommerce/menu/processor/MenuExpression.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.broadleafcommerce.menu.processor;
+
+import org.broadleafcommerce.common.web.expression.BroadleafVariableExpression;
+
+import java.util.Map;
+
+public interface MenuExpression extends BroadleafVariableExpression {
```
Comment (oleksiimiroshnyk) [inline src/main/java/org/broadleafcommerce/menu/processor/MenuExpression.java]: what is the purpose of this interface ?

File: src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java [hunk 3]
```diff
@@ -41,7 +39,7 @@
  */
 @Component("blMenuProcessor")
 @ConditionalOnTemplating
-public class MenuProcessor extends AbstractBroadleafVariableModifierProcessor {
+public class MenuProcessor implements MenuExpression {
```
Comment (oleksiimiroshnyk) [inline src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java]: guess we can implement BroadleafVariableExpression directly

File: src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java [hunk 4]
```diff
@@ -78,8 +73,8 @@ public Map<String, Object> populateModelVariables(String tagName, Map<String, St
 
         Map<String, Object> newModelVars = new HashMap<>();
         if (menu != null) {
-            newModelVars.put(resultVar, menuService.constructMenuItemDTOsForMenu(menu));
-            extensionManager.getProxy().addAdditionalFieldsToModel(tagName, tagAttributes, newModelVars, context);
+            newModelVars.put("menuName", menuService.constructMenuItemDTOsForMenu(menu));
```
Comment (oleksiimiroshnyk) [inline src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java]: why do we need this line ?

File: src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java [hunk 5]
```diff
@@ -63,10 +61,7 @@ public int getPrecedence() {
     }
 
     @Override
-    public Map<String, Object> populateModelVariables(String tagName, Map<String, String> tagAttributes, BroadleafTemplateContext context) {
-        String resultVar = tagAttributes.get("resultVar");
-        String menuName = tagAttributes.get("menuName");
-        String menuId = tagAttributes.get("menuId");
+    public Map<String, Object> getMenu(String menuId, String menuName) {
```
Comment (oleksiimiroshnyk) [inline src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java]: why do we need map as return type ? The idea of that method to get list of menu item dtos and put it under expected name in the map. In the new approach we don't use map we simply return what we ask for, in this case List<MenuItemDTO>

File: src/main/java/org/broadleafcommerce/menu/processor/MenuProcessorExtensionHandler.java [hunk 6]
```diff
@@ -34,6 +34,6 @@ public interface MenuProcessorExtensionHandler extends ExtensionHandler {
      *
      * @return - ExtensionResultStatusType
      */
-    public ExtensionResultStatusType addAdditionalFieldsToModel(String tagName, Map<String, String> tagAttributes, Map<String, Object> newModelVars, BroadleafTemplateContext context);
+    public ExtensionResultStatusType addAdditionalFieldsToModel(Map<String, Object> newModelVars, String menuId, String menuName);
```
Comment (oleksiimiroshnyk) [inline src/main/java/org/broadleafcommerce/menu/processor/MenuProcessorExtensionHandler.java]: idea of that extension is to allow some other modules override/modify result of execution of the method.
So I think signature should be something like 
public ExtensionResultStatusType addAdditionalFieldsToModel(List<MenuItemDTO> menuItems, String menuName);

or something like that... and extension will modify menuItems as needed

File: src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java (hunk 4 above)
Comment (oleksiimiroshnyk) [reply src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java]: guess we need something like List<MenuItemDTO>  menuItems = menuService.constructMenuItemDTOsForMenu(menu)

File: src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java [hunk 7]
```diff
@@ -78,8 +73,8 @@ public Map<String, Object> populateModelVariables(String tagName, Map<String, St
 
         Map<String, Object> newModelVars = new HashMap<>();
         if (menu != null) {
-            newModelVars.put(resultVar, menuService.constructMenuItemDTOsForMenu(menu));
-            extensionManager.getProxy().addAdditionalFieldsToModel(tagName, tagAttributes, newModelVars, context);
+            newModelVars.put("menuName", menuService.constructMenuItemDTOsForMenu(menu));
+            extensionManager.getProxy().addAdditionalFieldsToModel(newModelVars, menuId, menuName);
```
Comment (oleksiimiroshnyk) [inline src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java]: extensionManager.getProxy().addAdditionalFieldsToModel(menuItems, menuName);

File: src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java [hunk 8]
```diff
@@ -78,8 +73,8 @@ public Map<String, Object> populateModelVariables(String tagName, Map<String, St
...
 
             newModelVars.put("menuLinkedData", linkedDataService.getLinkedData(menu));
```
Comment (oleksiimiroshnyk) [inline src/main/java/org/broadleafcommerce/menu/processor/MenuProcessor.java]: just return menuItems

**Review by oleksiimiroshnyk — CHANGES_REQUESTED** [review]

**Review by oleksiimiroshnyk — APPROVED** [review]

**Pull request merged**